src/csharp/Microsoft.Spark/ML/Linalg/SparseVector.cs

```diff
@@ -0,0 +1,159 @@
+        public SparseVector(int length, int[] indices, double[] values) => 
+            _jvmObject = (JvmObjectReference)SparkEnvironment.JvmBridge.CallStaticJavaMethod(
+                    s_sparseVectorClassName, "sparse", length, indices, values);
+        //Note this calls Vectors.sparse as the SparseVector ctor is private
+
+        internal SparseVector(JvmObjectReference jvmObject) => _jvmObject = jvmObject;
+
+        JvmObjectReference IJvmObjectReferenceProvider.Reference => _jvmObject;
+
+        /// <summary>
+        /// Gets the value of the ith element.
+        /// </summary>
+        /// <param name="i">The element to return</param>
+        /// <returns>The specific value at the ith element or zero if it is not set</returns>
+        public double Apply(int i) => (double)_jvmObject.Invoke("apply", i);
+
+        /// <summary>
+        /// Number of active entries. An "active entry" is an element which is explicitly stored, regardless
+        /// of its value.
+        /// </summary>
+        /// <returns>The number of active entries in the <see cref="SparseVector"/></returns>
+        public int NumActives() => (int)_jvmObject.Invoke("numActives");
+
+        /// <summary>
+        /// Number of nonzero elements.
+        /// </summary>
+        /// <returns>The number of non zero elements in the <see cref="SparseVector"/></returns>
+        public int NumNonZeros() => (int)_jvmObject.Invoke("numNonzeros");
+
+        /// <summary>
+        /// Find the index of a maximal element. Returns the first maximal element in case of a tie.
+        /// Returns -1 if vector has length 0.
+        /// </summary>
+        /// <returns>The maximal element</returns>
+        public int ArgMax() => (int)_jvmObject.Invoke("argmax");
+
+        /// <summary>
+        /// The indices of the <see cref="SparseVector"/>.
+        /// </summary>
+        /// <returns>The indices of the <see cref="SparseVector"/></returns>
+        public int[] Indices() => (int[])_jvmObject.Invoke("indices");
+
+        /// <summary>
+        /// The values of the <see cref="SparseVector"/>.
+        /// </summary>
+        /// <returns>The values of the <see cref="SparseVector"/></returns>
+        public double[] Values() => (double[])_jvmObject.Invoke("values");
+
+        /// <summary>
+        /// Converts the instance to a double array. This will return every element of the
+        /// <see cref="SparseVector"/> whether it is a 0 or not. 
+        /// </summary>
+        /// <returns>Array of every element in the <see cref="SparseVector"/></returns>
+        public double[] ToArray() => (double[])_jvmObject.Invoke("toArray");
+
+        /// <summary>
+        /// Returns the JVM toString value rather than the .NET ToString default
+        /// </summary>
+        /// <returns>JVM toString() value</returns>
+        public override string ToString() => (string)_jvmObject.Invoke("toString");
+
```

Comment:
I see there a few methods we haven't exposed here such as `asML()`, `copy()`, `size()`, `toJson()`.. Is there a reason for this?

Reply:
A few reasons, the asMl - I didn’t implement the mlreader/mlwriter on any of the classes yet, was going to do a pr later on to back fill them all.

the copy, toJson, count etc I wasn’t sure if the whole approach to this with the udt was going to be acceptable so put in the minimum to make it useable - if this is all good then I’ve got DenseVector, SparseMatrix and DenseMatrix to add so will follow up with more pr’s.

Reply:
If you think they have to be there let me know and I’ll add them.